### PR TITLE
Adjust the Portable PDB validation code to account for missing VB fake locals

### PR DESCRIPTION
**Pull request by tmat**
Fixes https://github.com/dotnet/roslyn/issues/8494.

### CODE REVIEW

**Comment by tmat** [comment]
@amcasey @KevinH-MS

File: src/Test/Utilities/Shared/Pdb/PdbValidation.cs [hunk 1]
```diff
@@ -143,14 +143,20 @@ public static class PdbValidation
...
+                emptyScopes = (from e in pdb.DescendantsAndSelf()
+                               where e.Name == "scope" && !e.HasElements
+                               select e).ToArray();
+
+                foreach (var e in emptyScopes)
+                {
+                    e.Remove();
+                }
             }
+            while (emptyScopes.Any());
```
Comment (KevinH-MS) [inline src/Test/Utilities/Shared/Pdb/PdbValidation.cs]: Under what conditions do we repeat the ```do...while```?  (naively, it seems like we'll only ever execute this once)

Comment (tmat) [inline src/Test/Utilities/Shared/Pdb/PdbValidation.cs]:
```
<scope>
  <scope>
    <local name="$VB$ResumableLocal_foo" />
  </scope>
</scope>
```

Comment (KevinH-MS) [inline src/Test/Utilities/Shared/Pdb/PdbValidation.cs]: Does one of the tests above cover this?

Comment (KevinH-MS) [inline src/Test/Utilities/Shared/Pdb/PdbValidation.cs]: Nevermind...I guess ```PDBIteratorTests.vb```.

**Comment by KevinH-MS** [comment]
:+1:

File: src/Test/Utilities/Shared/Pdb/PdbValidation.cs [hunk 2]
```diff
@@ -143,14 +143,20 @@ public static class PdbValidation
...
-                              select e;
+            XElement[] emptyScopes;
 
-            foreach (var e in emptyScopes.ToArray())
+            do
             {
-                e.Remove();
+                emptyScopes = (from e in pdb.DescendantsAndSelf()
+                               where e.Name == "scope" && !e.HasElements
+                               select e).ToArray();
```
Comment (amcasey) [inline src/Test/Utilities/Shared/Pdb/PdbValidation.cs]: Why `ToArray`?

**Comment by amcasey** [comment]
:+1:

File: src/Test/Utilities/Shared/Pdb/PdbValidation.cs (hunk 2 above)
Comment (tmat) [inline src/Test/Utilities/Shared/Pdb/PdbValidation.cs]: So that we can modify the XML tree while enumerating it.

**Pull request merged**